### PR TITLE
[FIX] purchase: remove HTML tags from tax description in PO report

### PR DESCRIPTION
**Pull request by elhayyany**
When printing the purchase order, the tax values were displayed with HTML tags because the field is an HTML field, and it was passed to the `join` function, which treated it as plain text.
```<span t-esc="', '.join(map(lambda x: x.description or x.name, line.taxes_id))"/>```
To fix this, the tax values are now converted to plain text before being passed to the `join` function.

Steps to reproduce:
1. Create a purchase order PDF using taxes from a localization (e.g., Belgium).
2. Print the purchase order and observe the tax values displaying with HTML tags.

opw-4260595

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/purchase/models/purchase_order_line.py [hunk 1]
```diff
@@ -678,3 +678,7 @@ def _get_select_sellers_params(self):
         return {
             "order_id": self.order_id,
         }
+
+    def _get_cleaned_tax_names(self):
+        """ Return a comma-separated string of the taxes names, without HTML tags. """
+        return ', '.join(html2plaintext(tax.description or tax.name) for tax in self.taxes_id)
```
Comment (nle-odoo) [inline addons/purchase/models/purchase_order_line.py:684]: since `tax.name`  is a Char it seems it doesn't need `html2plaintext` also testing `html2plaintext` it returns an empty string for `<p></p>` or `<p><br/></p>`, so doing:

```suggestion
        return ', '.join(html2plaintext(tax.description) or tax.name for tax in self.taxes_id)
```
should have the advantage on fallbacking on name if description is an empty html tag.

**Review by nle-odoo — APPROVED** [review]
seems good to me

other way to do this would be to show HTML, but since we join by a , it doesn't seem to make sense to have things of different size / spacing / on several lines / ...

File: addons/purchase/report/purchase_order_templates.xml [hunk 2]
```diff
@@ -88,7 +88,7 @@
                                     <span class="text-align-bottom"><span t-field="line.discount"/>%</span>
                                 </td>
                                 <td class="text-end">
-                                    <span t-esc="', '.join(map(lambda x: x.description or x.name, line.taxes_id))"/>
```
Comment (ticodoo) [inline addons/purchase/report/purchase_order_templates.xml:91]: It seems like the `x.description` here was changed in this PR https://github.com/odoo/odoo/pull/160342 when it shouldn't have been. Can you please change it back to `x.invoice_label`? Or even better, back to the original `join([(tax.invoice_label or tax.name) for tax in line.taxes_id])`?

Comment (elhayyany) [reply addons/purchase/report/purchase_order_templates.xml:91]: hello, I'll update it

**Comment by ticodoo** [comment]
Thanks!

@robodoo r+
